java/src/main/java/io/process/analytics/tools/bpmn/generator/algo/ShapeLayouter.java

```diff
@@ -16,29 +16,64 @@
 
     public Grid layout(SortedDiagram diagram) {
         Grid grid = new Grid();
+        for (int i = 0; i < diagram.getShapes().size(); i++) {
+
+        }
         for (Shape shape : diagram.getShapes()) {
+            Position positionOfCurrentShape;
             List<Edge> incomingEdges = diagram.getIncomingEdges(shape.getId());
             if (incomingEdges.isEmpty()) {
                 //This is a start node, insert it in a new column
-                grid.add(position(shape, 0, grid.getLastColumnIndex() + 1));
+                positionOfCurrentShape = addStartShape(grid, shape);
             } else if (incomingEdges.size() == 1) {
                 //find the previous node position
                 String previousShapeID = incomingEdges.get(0).getFrom();
                 List<Edge> outgoingEdgesOfPreviousShape = diagram.getOutgoingEdges(previousShapeID);
                 if (outgoingEdgesOfPreviousShape.size() == 1) {
-                    addDirectlyNextTo(grid, shape, previousShapeID);
+                    positionOfCurrentShape = addDirectlyNextTo(grid, shape, previousShapeID);
                 } else {
-                    //next to a split
-                    throw new UnsupportedOperationException("Positioning a Node that is splitting is not yet supported");
+                    positionOfCurrentShape = addSplit(grid, shape, previousShapeID, outgoingEdgesOfPreviousShape);
                 }
             } else {
-                addJoin(grid, shape, incomingEdges);
+                positionOfCurrentShape = addJoin(grid, shape, incomingEdges);
+            }
+            List<Edge> outgoingEdges = diagram.getOutgoingEdges(shape.getId());
+            if (outgoingEdges.size() > 1) {
+                //add rows to place elements of this split
```

Comment:
:question: I may be messed by the comment about the split
at a 1st glance, this is strange to have something which is always run but which only relates to split
What I understood is that you always add rows to manage eventual splits. Is that correct?

Reply:
At first, I started adding rows when adding the elements after the split:
i.e. when calling the `addSplit` method, I was adding rows to handle all elements.
Finally, I found it easier to add the rows when we are processing the elements that have multiple outgoing transitions.
I did not understood that we always add rows to manage an eventual split. It's only written that we need to add rows to put all elements after a split. It's not written how we do it. 
We should check the implementation in oryx to see how they handled that.

Reply:
ok, thanks for the info
As it is working for now, we can live with it (at least it is not a problem to me)